### PR TITLE
[6.x] Provisional support for PHP 8.0

### PR DESCRIPTION
**Pull request by GrahamCampbell**
The purpose of this PR is to get the 6.x branch ready for PHP 8.0, without official support yet. Official support will come in https://github.com/laravel/framework/pull/33388 once all the outstanding issues are resolved. Most notably, Doctrine DBAL and Faker remain blockers, and possibly AWS SDK, since Amazon have made no effort here. All the effort has come from the OSS community on their dependencies.

### CODE REVIEW

**Comment by GrahamCampbell** [comment]
I have not added PHP 8.0 to the builds here, since it is not useful, in particular when merging upwards to Laravel 8 where composer installs will fail due to faker. We can track the PHP 8.0 test status over on the other PR, when we rebase.

**Review by driesvints — APPROVED** [review]

File: src/Illuminate/Container/BoundMethod.php [hunk 1]
```diff
@@ -28,9 +28,7 @@ public static function call($container, $callback, array $parameters = [], $defa
         }
 
         return static::callBoundMethod($container, $callback, function () use ($container, $callback, $parameters) {
-            return call_user_func_array(
-                $callback, static::getMethodDependencies($container, $callback, $parameters)
-            );
+            return $callback(...static::getMethodDependencies($container, $callback, $parameters));
```
Comment (stancl) [inline src/Illuminate/Container/BoundMethod.php:31]: This is a breaking change.

```
@livewire('my-component', ['foo' => 'bar'])
```

makes a call where the return value of the `getMethodDependencies()` method is an array with string keys.

Resulting in an exception:

> Cannot unpack array with string keys

Comment (driesvints) [reply src/Illuminate/Container/BoundMethod.php:31]: Can you please open an issue so we can focus the discussion?

File: composer.json [hunk 2]
```diff
@@ -20,13 +20,13 @@
         "ext-mbstring": "*",
         "ext-openssl": "*",
         "doctrine/inflector": "^1.4|^2.0",
-        "dragonmantank/cron-expression": "^2.0",
+        "dragonmantank/cron-expression": "^2.3.1",
         "egulias/email-validator": "^2.1.10",
         "league/commonmark": "^1.3",
-        "league/flysystem": "^1.0.34",
+        "league/flysystem": "^1.1",
```
Comment (bytestream) [inline composer.json:26]: This bumps the minimum supported version from `7.2.0` to `7.2.5`. https://laravel.com/docs/6.x states `>= 7.2.0`

Arguably BC break but I'm sure many would disagree.

Comment (stancl) [reply composer.json:26]: I think you only get this update by running `composer update` so the dependencies would get upgraded anyway?

Though maybe BC if you selectively upgrade dependencies.

Comment (driesvints) [reply composer.json:26]: Also: bumping minimum versions in a stable release isn't a BC break: https://semver.org/#what-should-i-do-if-i-update-my-own-dependencies-without-changing-the-public-api

Comment (bytestream) [reply composer.json:26]: The main point of the documentation is now wrong

Comment (driesvints) [reply composer.json:26]: @bytestream what main point?

**Pull request merged**